### PR TITLE
Update test_accuracy.py to generate data with different random seed

### PR DESCRIPTION
**Pull request by puhuk**
…er rank

Fixes #{issue number}

Description: Change generating data logic from slicing to different random seed per rank in `_test_distrib_integration_multiclass`
After this PR, will correct `_test_distrib_integration_multilabel` and `_test_distrib_integration_list_of_tensors_or_numbers`

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/metrics/test_accuracy.py [hunk 1]
```diff
@@ -366,18 +366,23 @@ def _test(n_epochs, metric_device):
         s = 16
         n_classes = 10
 
+        torch.manual_seed(12 + rank)
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_accuracy.py:369]: Line 361 already contains `torch.manual_seed(12)`, let's use it once somewhere.

File: tests/ignite/metrics/test_accuracy.py [hunk 2]
```diff
@@ -366,18 +366,23 @@ def _test(n_epochs, metric_device):
         s = 16
         n_classes = 10
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_accuracy.py:367]: Let's rename `s` variable for better code comprehension

File: tests/ignite/metrics/test_accuracy.py [hunk 3]
```diff
@@ -366,18 +366,23 @@ def _test(n_epochs, metric_device):
...
 
+        torch.manual_seed(12 + rank)
+
         offset = n_iters * s
```
Comment (vfdev-5) [inline tests/ignite/metrics/test_accuracy.py:371]: Same for `offset`, let's rename it or even remove it

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @puhuk 
Please fix code formatting issue: https://github.com/pytorch/ignite/runs/7781175697?check_suite_focus=true#step:11:84

File: tests/ignite/metrics/test_accuracy.py (hunk 3 above)
Comment (puhuk) [reply tests/ignite/metrics/test_accuracy.py]: I change `s` to `batch_size` and eliminate `offset`

**Pull request merged**